synapse/rest/admin/users.py

```diff
@@ -355,13 +390,18 @@ async def on_PUT(
             if password is not None:
                 password_hash = await self.auth_handler.hash(password)
 
+            new_user_approved = False
+            if self._msc3866_enabled and approved is True:
+                new_user_approved = True
```

Comment:
```suggestion
            new_user_approved = self._msc3866_enabled and approved is True
```

Reply:
Worth noting that we are consciously requiring even admins to specify `approved = True` rather than assuming that an admin creating the account at all is sufficient for approval by default.

Should this either:
- just approve the user unless overridden
- not depend on whether `require_approval_for_new_accounts` is True?

Thinking about it: if we unlock this MSC as-is, the current implementation will break existing admin API uses.

Reply:
> Worth noting that we are consciously requiring even admins to specify `approved = True` rather than assuming that an admin creating the account at all is sufficient for approval by default.

Thanks for pointing that out, that's a thinko from me. Users created by the admin API should be considered approved. Plus the condition for `new_user_approved` meant that if the feature isn't enabled new users are considered not approved, which is incorrect.

Reply:
So

> just approve the user unless overridden

Is the correct behaviour.